controllers/binding/snapshotenvironmentbinding_adapter.go

```diff
@@ -148,7 +167,7 @@ func (a *Adapter) EnsureEphemeralEnvironmentsCleanedUp() (controller.OperationRe
 	a.logger.Info("The SnapshotEnvironmentBinding encountered an issue deploying snapshot on ephemeral environments",
 		"snapshotEnvironmentBinding.Name", a.snapshotEnvironmentBinding.Name,
 		"message", snapshotErrorMessage)
-	_, err := gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, snapshotErrorMessage)
+	_, err = gitops.MarkSnapshotAsFailed(a.client, a.context, a.snapshot, snapshotErrorMessage)
```

Comment:
Would you like to add the [check](https://github.com/redhat-appstudio/integration-service/blob/d768469006ad27feeb46e9d534f17e3d4d5eef57/controllers/integrationpipeline/integrationpipeline_adapter.go#L151C3-L151C128) before MakSnapshotAsFailed ?  Otherwise we may need another PR to fix it.

Reply:
It's not related to this feature, I can add it in separate PR

Reply:
https://github.com/redhat-appstudio/integration-service/pull/321 PR opened here